### PR TITLE
Use Recognizer library only with Maybe monad

### PR DESCRIPTION
**Pull request by brianhuffman**
This PR prepares saw-script for some planned changes to the saw-core `Recognizer` library. Specifically, I plan to restrict `Recognizer` to work only with the `Maybe` monad, so that we can avoid the possibility of calling `fail` in an arbitrary monad.

### CODE REVIEW

File: src/SAWScript/JavaBuiltins.hs [hunk 1]
```diff
@@ -22,6 +22,7 @@ import Control.Lens
 import Control.Monad.State
 import Control.Monad.Trans.Except
 import Data.List (partition)
+import Data.Maybe (mapMaybe)
```
Comment (andreistefanescu) [inline src/SAWScript/JavaBuiltins.hs]: nit: I would import the entire module. This style of imports leads to unnecessary changes any time the code changes and uses something else from the same module

Comment (brianhuffman) [reply src/SAWScript/JavaBuiltins.hs]: There's a tradeoff here. Explicit import lists and/or qualified imports make it nicer for anyone reading the code, because they make it easy to figure out where any particular function comes from. Although I suppose it's reasonable to make the other choice for well-known modules in base like `Data.List`, `Data.Maybe`, or `Control.Monad`.

I'll get rid of the import list the next time I have to change either of those lines :)
